addons/sale_project/models/project.py

```diff
@@ -394,21 +396,60 @@ def _get_revenues_items_from_sol(self, domain=None, with_action=True):
+        revenue_items_from_sol, sol_ids = self._get_revenues_items_from_sol(
             [('order_id', 'in', self.sudo()._get_sale_orders().ids)],
             with_action,
         )
         revenues = profitability_items['revenues']
         revenues['data'] += revenue_items_from_sol['data']
         revenues['total']['to_invoice'] += revenue_items_from_sol['total']['to_invoice']
         revenues['total']['invoiced'] += revenue_items_from_sol['total']['invoiced']
+        # calculate the revenues from invoices without sale orders
+        accounted_invoice_lines_ids = self.env['sale.order.line'].browse(sol_ids).invoice_lines.ids
+        query = self.env['account.move.line'].sudo()._search([
+            ('analytic_account_id', 'in', self.analytic_account_id.ids),
+            ('move_id.move_type', 'in', ['out_invoice', 'out_refund']),
+            ('parent_state', 'in', ['draft', 'posted']),
+            ('price_subtotal', '>', 0),
+            ('id', 'not in', accounted_invoice_lines_ids)
+        ])
+        # account_move_line__move_id is the alias of the joined table account_move in the query
+        # we can use it, because of the "move_id.move_type" clause in the domain of the query, which generates the join
+        # this is faster than a search_read followed by a browse on the move_id to retrieve the move_type of each account.move.line
+        query_string, query_param = query.select('price_subtotal', 'parent_state', 'account_move_line__move_id.move_type')
+        self._cr.execute(query_string, query_param)
+        invoices_move_line_read = [moves for moves in self._cr.dictfetchall()]
+        if invoices_move_line_read:
+            amount_invoiced = amount_to_invoice = 0.0
+            for moves_read in invoices_move_line_read:
+                if moves_read['parent_state'] == 'draft':
+                    if moves_read['move_type'] == 'out_invoice':
+                        amount_to_invoice += moves_read['price_subtotal']
+                    else:  # moves_read['move_type'] == 'out_refund'
+                        amount_to_invoice -= moves_read['price_subtotal']
+                else:  # moves_read['parent_state'] == 'posted'
+                    if moves_read['move_type'] == 'out_invoice':
+                        amount_invoiced += moves_read['price_subtotal']
+                    else:  # moves_read['move_type'] == 'out_refund'
+                        amount_invoiced -= moves_read['price_subtotal']
+            section_id = 'other_invoice_revenues'
+            invoices_revenues = {
+                'id': section_id,
+                'sequence': self._get_profitability_sequence_per_invoice_type()[section_id],
+                'invoiced': amount_invoiced,
+                'to_invoice': amount_to_invoice,
+            }
+            revenues['data'].append(invoices_revenues)
+            revenues['total']['invoiced'] += amount_invoiced
+            revenues['total']['to_invoice'] += amount_to_invoice
```

Comment:
I am wondering if it is not better to add this code in the method above (`_get_revenues_items_from_sol`), I agree the name of method does not really fit with your code but it will be easier for you? 🤔 

What do you think? 🙂

Reply:
Might be missleading to the reader, but if `_get_profitability_items` is too large, I could refactor that code into a private function named `_get_revenue_items_from_invoices_without_so`, or `_get_revenue_items_from_invoices(excluded_move_lines_ids=False)` or something along those lines.

Reply:
We could do it and call it in `_get_revenues_items_from_sol` for instance to avoid changing the signature of the method. 🤔 

What do you think? 🙂

Reply:
If we follow the policy of "don't change the function signature because of possible client's custom code that may break" I couldn't put the code inside `_get_revenues_items_from_sol`, because I would need to change the return type (a list of dict instead of a dict), because the aggregation in the profitability_items is done at the call-site, not inside the function.
I've refactored the code inside it's own function, and to keep the signature I was forced to redo the query on the `sale.order`.